src/frameutils/frame.py

```diff
@@ -0,0 +1,146 @@
+import asyncio
+from typing import Optional
+from .bluetooth import Bluetooth
+from .files import FrameFileSystem
+from .camera import Camera
+import random
+import re
+
+class Frame:
+    """Entrypoint to the Frame SDK"""
+    
+    bluetooth : Bluetooth = None
+    files : FrameFileSystem = None
+    camera : Camera = None
+    
+    def __init__(self):
+        self.bluetooth = Bluetooth()
+        self.files = FrameFileSystem(self)
+        self.camera = Camera(self)
+        
+    async def __aenter__(self):
+        await self.ensure_connected()
+        return self
+    
+    async def __aexit__(self, exc_type, exc_value, traceback):
+        if self.bluetooth.is_connected():
+            await self.bluetooth.disconnect()
+        
+    async def ensure_connected(self):
+        """Ensure the Frame is connected, establishing a connection if not"""
+        if not self.bluetooth.is_connected():
+            await self.bluetooth.connect()
+            await self.inject_all_library_functions()
```

Comment:
Are you sure?  Seems best to let the exceptions bubble up since they are descriptive of the issue and we don't want to continue if we were unable to connect.